From 4bd520cf77585674c8c4c97b8956509f510940cd Mon Sep 17 00:00:00 2001
From: Mark Enriquez <enriquezmark36@gmail.com>
Date: Sun, 5 Apr 2020 09:27:18 +0800
Subject: [PATCH 01/12] libcamera: scx15: Apply SENSOR_IO_POWER_ONOFF on
 Sensor_PowerOn_Ex

This requires the patch named as
"libcamera: scx15: IOCTL extensions for some SPRD devices"

In Week 3 of the World-wide Quarantine due to the Coronavirus pandemic,

it seems that the SENSOR_IO_POWER_ONOFF was also called from the Sensor_PowerOn_Ex()
function not just the plain 'ol PowerOn function. Though this seems a useless
observation, this is a large part of the _cause_ why the autofocus on the
s5k4ec sensor fails.

Change-Id: I16e34acee379c254bdc1966fd9dd2833d8e36a0f
Signed-off-by: Mark Enriquez <enriquezmark36@gmail.com>
---
 libcamera/scx15/sensor_drv_u/src/sensor_drv_u.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libcamera/scx15/sensor_drv_u/src/sensor_drv_u.c b/libcamera/scx15/sensor_drv_u/src/sensor_drv_u.c
index 1cd5a3d..bac8011 100644
--- a/libcamera/scx15/sensor_drv_u/src/sensor_drv_u.c
+++ b/libcamera/scx15/sensor_drv_u/src/sensor_drv_u.c
@@ -827,6 +827,9 @@ LOCAL void Sensor_PowerOn_Ex(uint32_t sensor_id)
 
 	if (PNULL != power_func) {
 		power_func(1);
+#if defined(CONFIG_CAMERA_IOCTL_IOCTL_HAS_POWER_ONOFF)
+		_Sensor_Device_PowerON(1);
+#endif
 	} else {
 		Sensor_PowerDown(power_down);
 		Sensor_SetVoltage(dvdd_val, avdd_val, iovdd_val);
-- 
2.27.0

